src/wp-includes/media.php

```diff
@@ -1137,8 +1137,12 @@ function wp_get_attachment_image( $attachment_id, $size = 'thumbnail', $icon = f
 			}
 		}
 
+		/** This filter is documented in wp-includes/media.php */
+		$add_auto_sizes = apply_filters( 'wp_img_tag_add_auto_sizes', true, '' );
```

Comment:
It's not great to pass an empty string here. Obviously we don't have a full image tag available here, but that means the filter would be cumbersome to use if developers can't rely on there being an HTML string in this parameter.

I could see two alternatives here for the short-term:
* We could for now make this a simple on/off filter, without any context, both here and in `wp_img_tag_add_auto_sizes()`. This would be similar to how `wp_lazy_loading_enabled` works. Given the main reason for the filter for now is to "emergency turn off" the feature in case of a problem, I think that's a reasonable short-term solution. We could still explore granular filtering options per image in the 6.8 cycle without being in a rush.
* Or we don't use this filter here, but keep it in `wp_img_tag_add_auto_sizes()`, but then in both places apply a context-less filter as a general on/off switch. This would be similar to how lazy-loading filters work: There is `wp_lazy_loading_enabled` as on/off switch, and `wp_img_tag_add_loading_attr`, which is intended for a specific content image.

I think the first option would be simpler, sufficient for what's needed as a hot fix, and more future-proof, so I'd prefer that. We could then expand the filtering for 6.8.

Reply:
I noticed that too. I'd opt for removing the image context entirely if we're not consolidating this logic all in the `wp_img_tag_add_auto_sizes()` function and consider adding it as an enhancement.

Reply:
Removed in [214f91d](https://github.com/WordPress/wordpress-develop/pull/7812/commits/214f91da913ccfe3630cbacf807862ab7cfc073c).

I'll make one last case for why I think we should reconsider consolidating this logic to `wp_img_tag_add_auto_sizes()` as was experimented in [3cfcbe2](https://github.com/WordPress/wordpress-develop/pull/7812/commits/3cfcbe2052e5e89af0eb88aed2441876c18b5e9f):

1. The consolidation allows the image context to be consistently passed to the filter, making it more useful.
2. It makes the filter name, `wp_img_tag_add_auto_sizes` make more sense, because it matches the name of the function responsible for the logic.
3. It avoids edge cases where changes to the markup from `wp_get_attachment_image_attributes` or `wp_get_attachment_image` are not considered before auto-sizes gets applied.

The main downside that I see for _not_ consolidating this logic in one place is that it means adding the need to parse the finalized HTML again to inspect the attributes. I'm not convinced this alone outweighs the benefits of consolidation.

Reply:
I think there are ways to consolidate the logic without the downside of parsing the HTML again, but this is something that's better done as part of a 6.8 enhancement.